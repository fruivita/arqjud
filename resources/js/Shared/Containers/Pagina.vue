<!--
    Container para o conteúdo propriamente dito da página.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<script setup>
defineProps({
    titulo: { type: String },
});
</script>

<template>
    <article class="space-y-12 py-3">
        <h1 v-if="titulo" class="text-center text-2xl font-bold" dusk="titulo">
            {{ titulo }}
        </h1>

        <div class="space-y-12" dusk="slot">
            <slot></slot>
        </div>
    </article>
</template>
