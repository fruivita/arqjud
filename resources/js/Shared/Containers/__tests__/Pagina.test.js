/**
 * Testes para o componente Pagina.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import Pagina from '@/Shared/Containers/Pagina.vue';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    mountFunction = (options = {}) => {
        return mount(Pagina, { ...options });
    };
});

describe('Pagina', () => {
    // Caminho feliz
    test('propriedades do componente estão definidas', () => {
        expect(Pagina.props).toMatchObject({
            titulo: { type: String },
        });
    });

    test('renderiza o componente sem o título respeitando o snapshot', () => {
        expect(mountFunction().html()).toMatchSnapshot();
    });

    test('renderiza o componente com o título respeitando o snapshot', () => {
        const wrapper = mountFunction({ props: { titulo: 'foo' } });

        expect(wrapper.html()).toMatchSnapshot();
    });

    test('renderiza o componente com slot respeitando o snapshot', () => {
        const wrapper = mountFunction({ slots: { default: 'foo' } });

        expect(wrapper.html()).toMatchSnapshot();
    });
});
