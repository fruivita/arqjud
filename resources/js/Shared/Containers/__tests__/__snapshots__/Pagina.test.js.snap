// Vitest Snapshot v1

exports[`Pagina > renderiza o componente com o título respeitando o snapshot 1`] = `
"<article class=\\"space-y-12 py-3\\">
  <h1 class=\\"text-center text-2xl font-bold\\" dusk=\\"titulo\\">foo</h1>
  <div class=\\"space-y-12\\" dusk=\\"slot\\"></div>
</article>"
`;

exports[`Pagina > renderiza o componente com slot respeitando o snapshot 1`] = `
"<article class=\\"space-y-12 py-3\\">
  <!--v-if-->
  <div class=\\"space-y-12\\" dusk=\\"slot\\">foo</div>
</article>"
`;

exports[`Pagina > renderiza o componente sem o título respeitando o snapshot 1`] = `
"<article class=\\"space-y-12 py-3\\">
  <!--v-if-->
  <div class=\\"space-y-12\\" dusk=\\"slot\\"></div>
</article>"
`;
